nipype/interfaces/traits_extension.py

```diff
@@ -33,7 +33,7 @@
 from traits.api import BaseUnicode
 from traits.api import Unicode
 
-DictStrStr = traits.Dict(str, (bytes, str))
+DictStrStr = traits.Dict((bytes,str,unicode), (bytes,str,unicode))
```

Comment:
Apparently, the problem is in the keys of environ, not the values.

Did it work for you using ```DictStrStr = traits.Dict((bytes, str), (bytes,str))``` instead the version with ```unicode```?

(i.e. adding bytes for the validation of keys in DictStrStr)

This should do (it worked on the SGE environment I created for testing, where I could reproduce the issue)